testing/oracle/integration_test.go

```diff
@@ -0,0 +1,194 @@
+		databaseAdmin.Close()
+	}
+}
+
+func dropDatabase(t *testing.T, dbURI string) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	// Drop the testing database.
+	if err := databaseAdmin.DropDatabase(ctx, &databasepb.DropDatabaseRequest{Database: dbURI}); err != nil {
+		t.Fatalf("failed to drop testing database %v: %v", dbURI, err)
+	}
+}
+
+func prepareIntegrationTest(t *testing.T) string {
+	if databaseAdmin == nil {
+		t.Skip("Integration tests skipped")
+	}
+	tmpdir, err := ioutil.TempDir(".", "int-test-")
+	if err != nil {
+		log.Fatal(err)
+	}
+	return tmpdir
+}
+
+func TestIntegration_ORACLE_SchemaSubcommand(t *testing.T) {
+	onlyRunForEmulatorTest(t)
+	t.Parallel()
+	tmpdir := prepareIntegrationTest(t)
+	defer os.RemoveAll(tmpdir)
+	filePrefix := filepath.Join(tmpdir, "Oracle_IntTest.")
+
+	args := fmt.Sprintf("schema -prefix %s -source=%s -source-profile='host=localhost,user=STI,db_name=xe,password=test1,port=1521'", filePrefix, constants.ORACLE)
+	err := common.RunCommand(args, projectID)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+func TestIntegration_ORACLE_SchemaAndDataSubcommand(t *testing.T) {
+	onlyRunForEmulatorTest(t)
+	tmpdir := prepareIntegrationTest(t)
+	defer os.RemoveAll(tmpdir)
+
+	dbName := "schema-and-data"
+	dbURI := fmt.Sprintf("projects/%s/instances/%s/databases/%s", projectID, instanceID, dbName)
+	filePrefix := filepath.Join(tmpdir, "Oracle_IntTest.")
+
+	args := fmt.Sprintf("schema-and-data -prefix %s -source=%s  -source-profile='host=localhost,user=STI,db_name=xe,password=test1,port=1521' -target-profile='instance=%s,dbname=%s'", filePrefix, constants.ORACLE, instanceID, dbName)
+	err := common.RunCommand(args, projectID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer dropDatabase(t, dbURI)
+	checkResults(t, dbURI)
+}
+func checkResults(t *testing.T, dbURI string) {
+	// Make a query to check results.
+	client, err := spanner.NewClient(ctx, dbURI)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+
+	checkCommonDataType(ctx, t, client)
+}
+
+func checkCommonDataType(ctx context.Context, t *testing.T, client *spanner.Client) {
+	var date spanner.NullDate
+	var floatVal float64
+	var intVal int64
+	var numericVal big.Rat
+	var stringVal string
+	var timeVal spanner.NullTime
+	iter := client.Single().Read(ctx, "ALLTYPES", spanner.Key{3}, []string{"DATE_T", "FLOAT_T", "INTEGER_T", "NUMERIC_T", "VARCHAR_T", "TIMESTAMP_T"})
+	defer iter.Stop()
+	for {
+		row, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			t.Fatal(err)
+		}
+		if err := row.Columns(&date, &floatVal, &intVal, &numericVal, &stringVal, &timeVal); err != nil {
+			t.Fatal(err)
+		}
+	}
+	fmt.Fprintf(os.Stdout, "%v,%v,%v,%v,%v,%v", date, floatVal, intVal, numericVal, stringVal, timeVal)
+	if got, want := date.String(), "2022-01-18"; got != want {
+		t.Fatalf("Date are not correct: got %v, want %v", got, want)
+	}
+	if got, want := floatVal, 1234.56789; got != want {
```

Comment:
Consider converting these ifs to this alternate way of checking using struct comparison.
https://github.com/cloudspannerecosystem/harbourbridge/blob/master/testing/dynamodb/integration_test.go#L244

Reply:
Yes, updated accordingly.